cmd/shell.go

```diff
@@ -302,6 +309,44 @@ func LoadCert(path string) (cert []byte, err error) {
 	return
 }
 
+var ocMu sync.Mutex
+var openConnections int64
+
+// HandlerTimer monitors HTTP performance and sends the details to StatsD.
+func HandlerTimer(handler http.Handler, stats statsd.Statter, prefix string) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		stats.Inc(fmt.Sprintf("%s.HTTP.Rate", prefix), 1, 1.0)
+		ocMu.Lock()
+		openConnections++
```

Comment:
This could just be a sync/atomic.AddUint32 and then we can skip out on the locks everywhere.